contracts/EscrowedNori.sol

```diff
@@ -0,0 +1,725 @@
+    if (atTime >= escrowSchedule.endTime) {
+      return escrowSchedule.currentAmount;
+    }
+    uint256 rampTotalTime = escrowSchedule.endTime - escrowSchedule.startTime;
+    return
+      atTime < escrowSchedule.startTime
+        ? 0
+        : (escrowSchedule.currentAmount * (atTime - escrowSchedule.startTime)) /
+          rampTotalTime;
+  }
+
+  /**
+   * @notice Released balance less any claimed amount at `atTime` (implementation)
+   *  TODO is this even right anymore?
+   * @dev If any tokens have been revoked then the schedule (which doesn't get updated) may return more than the total
+   * escrow schedule amount. This is done to preserve the behavior of the escrow schedule despite a reduction in the
+   * total quantity of tokens releasing.
+   * i.o.w The rate of release does not change after calling `revokeUnreleasedTokens`
+   */
+  function _claimableBalanceOf(address account, uint256 atTime)
+    internal
+    view
+    returns (uint256)
+  {
+    uint256 balance = this.balanceOf(account); // default to this if escrow schedules don't exist?
+    uint256 totalReleasedBalance = 0;
+    uint256[] memory escrowScheduleIds = _addressToEscrowScheduleIds[account];
+    if (escrowScheduleIds.length > 0) {
+      for (uint256 i = 0; i < escrowScheduleIds.length; i++) {
+        totalReleasedBalance += _claimableBalanceOfSingleEscrowSchedule(
+          account,
+          escrowScheduleIds[i],
+          atTime
+        );
+      }
+      balance = totalReleasedBalance;
+    }
+    return balance;
+  }
+
+  /**
+   * @notice Released balance less any claimed amount at `atTime` for a single escrow schedule (implementation)
+   */
+  function _claimableBalanceOfSingleEscrowSchedule(
+    address account,
+    uint256 escrowScheduleId,
+    uint256 atTime
+  ) internal view returns (uint256) {
+    EscrowSchedule storage escrowSchedule = _addressToEscrowSchedules[account][
+      escrowScheduleId
+    ];
+    require(escrowSchedule.exists, "eNORI: no escrow schedule exists");
+    return
+      _releasedBalanceOfSingleEscrowSchedule(
+        account,
+        escrowScheduleId,
+        atTime
+      ) - escrowSchedule.claimedAmount;
+  }
+
+  /**
+   * @notice Released balance at `atTime` for a single escrow schedule
+   */
+  function _releasedBalanceOfSingleEscrowSchedule(
+    address account,
+    uint256 escrowScheduleId,
+    uint256 atTime
+  ) internal view returns (uint256) {
+    EscrowSchedule storage escrowSchedule = _addressToEscrowSchedules[account][
+      escrowScheduleId
+    ];
+    return
+      MathUpgradeable.max(
+        _linearReleaseAmountAvailable(escrowSchedule, atTime),
+        escrowSchedule.releasedAmountFloor
+      );
+  }
+
+  function _beforeOperatorChange(address, uint256) internal pure override {
+    revert("eNORI: operator actions disabled");
+  }
+
+  function send(
+    address,
+    uint256,
+    bytes memory
+  ) public pure override {
+    revert("eNORI: send disabled");
```

Comment:
definitely want to look into enabling these for this contract for defi compatibility